### PR TITLE
Clean up MFE/Redux usage

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Forked from [muselesscreator/redux](https://github.com/muselesscreator/frontend-app-gradebook/tree/redux)

JIRA: Part of [EDUCATOR-5264](https://openedx.atlassian.net/browse/EDUCATOR-5264)

**What changed?**

- Update selectors to better reflect the values needed by components.
- Fix adjacent ESLint issues (e.g. import orders, export style, commas, whitespace)

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [x] Bumped version number [package.json](../package.json)

**Testing Instructions**

- No functional changes introduced, visual code review and smoke tests pass

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx [hunk 1]
```diff
@@ -9,12 +10,20 @@ import {
...
 } from '.';
 
-jest.mock('data/selectors/filters', () => ({
+jest.mock('data/selectors', () => ({
   /** Mocking to use passed state for validation purposes */
-  selectableAssignmentLabels: jest.fn().mockImplementation((state) => ({
-    state,
-    selectableLabels: 'selectableLabels',
-  })),
+  filters: {
```
Comment (nsprenkle) [inline src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx:15]: These values are arbitrary but should differ from each other to catch bad linking of filters to props

File: src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx [hunk 2]
```diff
@@ -103,51 +112,47 @@ describe('AssignmentFilter', () => {
       },
     };
     describe('assignmentFilterOptions', () => {
-      it('is drawn from selectableAssignmentLabels', () => {
```
Comment (nsprenkle) [inline src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx:106]: Deleted as the selectors now own the logic for how to map this data

File: src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx (hunk 1 above)
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx:15]: ++

**Review by muselesscreator — APPROVED** [review]
:-)

File: src/data/selectors/filters.js [hunk 3]
```diff
@@ -1,38 +1,73 @@
...
+    'assignmentType',
+    'cohort',
+    'courseGradeMax',
+    'courseGradeMin',
+    'track',
+    'includeCourseRoleMembers',
+  ],
+);
+const selectedAssignmentId = (state) => (simpleSelectors.assignment(state) || {}).id;
+const selectedAssignmentLabel = (state) => (simpleSelectors.assignment(state) || {}).label;
```
Comment (jansenk) [inline src/data/selectors/filters.js:58]: this seems like a fair amount of kinda confusing code for ... not much of a gain?

Comment (nsprenkle) [reply src/data/selectors/filters.js:58]: I think that's to make sure we don't try to access `id`/`label` on an undefined which would throw an exception. @muselesscreator?
